src/compiler/crystal/codegen/codegen.cr

```diff
@@ -1530,10 +1531,13 @@ module Crystal
     end
 
     def check_proc_is_not_closure(value, type)
-      check_fun_name = "~check_proc_is_not_closure"
-      func = @main_mod.functions[check_fun_name]? || create_check_proc_is_not_closure_fun(check_fun_name)
-      func = check_main_fun check_fun_name, func
-      value = call func, [value] of LLVM::Value
+      if value.type == llvm_typer.proc_type
+        check_fun_name = "~check_proc_is_not_closure"
+        func = @main_mod.functions[check_fun_name]? || create_check_proc_is_not_closure_fun(check_fun_name)
+        func = check_main_fun check_fun_name, func
+        proc_value = bit_cast value, llvm_typer.proc_type
```

Comment:
Is this extra bit case needed if we already checked `value.type`?

Reply:
Yup, sorry. Totally missed that out from an earlier experiment, good catch.